circleguard/gui.py

```diff
@@ -286,71 +313,345 @@ def scroll_to_bottom(self):
+        if id_ in self.loadable_ids:
+            return
+        self.loadable_ids.append(id_)
+        self.layout.addWidget(DragWidget(name, id_, dragging=False))
+
+
+
+
+class CheckW(QFrame):
+    remove_check_signal = pyqtSignal(int) # check id
+    ID = 0
+
+    def __init__(self, name):
+        super().__init__()
+        CheckW.ID += 1
+        # so we get the DropEvent
+        # https://doc.qt.io/qt-5/qdropevent.html#details
+        self.setAcceptDrops(True)
+
+        self.name = name
+        self.check_id = CheckW.ID
+        # will have LoadableW objects added to it once this Check is
+        # run by the main tab run button, so that cg.Check objects can
+        # be easily constructed with loadables
+        self.loadables = []
+
+        self.layout = QGridLayout()
+
+        self.drop_area = DropArea()
+        self.delete_button = QPushButton(self)
+        self.delete_button.setIcon(QIcon(str(resource_path("./resources/delete.png"))))
+        self.delete_button.setMaximumWidth(30)
+        self.delete_button.clicked.connect(partial(lambda check_id: self.remove_check_signal.emit(check_id), self.check_id))
+        title = QLabel()
+        title.setText(f"{name}")
+        self.layout.addWidget(title, 0, 0, 1, 7)
+        self.layout.addWidget(self.delete_button, 0, 7, 1, 1)
+        self.layout.addWidget(self.drop_area, 1, 0, 1, 8)
+        self.setLayout(self.layout)
+
+
+class StealCheckW(CheckW):
+    def __init__(self):
+        super().__init__("Replay Stealing Check")
+
+
+class RelaxCheckW(CheckW):
+    def __init__(self):
+        super().__init__("Relax Check")
+
+
+class CorrectionCheckW(CheckW):
+    def __init__(self):
+        super().__init__("Aim Correction Check")
+
+
+class DragWidget(QFrame):
+    """
+    A widget not meant to be displayed, but rendered into a pixmap with
+    #grab and stuck onto a QDrag with setPixmap to give the illusion of
+    dragging another widget.
+    """
+    remove_signal = pyqtSignal(int)  # drag id
+
+    def __init__(self, name, id, dragging=True):
+        super().__init__()
+        self.name = name
+        self.id = id
+        self.text = QLabel(name + f" (id: {self.id})")
+        layout = QGridLayout()
+        layout.addWidget(self.text, 0, 0, 1, 7)
+        if not dragging:
+            self.delete_button = QPushButton(self)
+            self.delete_button.setIcon(QIcon(str(resource_path("./resources/delete.png"))))
+            self.delete_button.setMaximumWidth(30)
+            self.delete_button.clicked.connect(partial(lambda check_id: self.remove_signal.emit(check_id), self.id))
+            layout.addWidget(self.delete_button, 0, 7, 1, 1)
+        self.setLayout(layout)
+
+
+class LoadableW(QFrame):
+    """
+    A widget representing a circleguard.Loadable, which can be dragged onto
+    a DetectWidget. Keeps track of how many LoadableWidgets have been created
+    as a static ID attribute.
+
+    W standing for widget.
```

Comment:
...
sounds unnecessary

Reply:
might be obvious to you and me right now, but I don't think it will be obvious if I come back to this code in three months. `W` isn't exactly an unambiguous acronym

Reply:
Class is called `LoadableW`, the documentation calls it `LoadableWidgets`. You should be able to understand that `LoadableWidgets` means that the class is a widget.

Reply:
that's true, as written the bottom line is unnecessary. But "how many LoadableWidgets" is outdated and should be changed to "how many LoadableWs" anyway, in which case an explicit explanation for the `W` suffix is appropriate I think

Reply:
Yea, if you change it to `LoadableW`s I agree with keeping the explanation.